kevlar/call.py

```diff
@@ -161,79 +170,128 @@ def call_snv(target, query, offset, length, ksize):
         window = q[minpos:maxpos]
         refrwindow = t[minpos:maxpos]
 
-        numoverlappingkmers = len(window) - ksize + 1
-        kmers = [window[i:i+ksize] for i in range(numoverlappingkmers)]
+        # numoverlappingkmers = len(window) - ksize + 1
+        # kmers = [window[i:i+ksize] for i in range(numoverlappingkmers)]
         refr = diff[1].upper()
         alt = diff[2].upper()
-        localcoord = offset + diff[0]
+        localcoord = diff[0]
+        if not targetshort:
+            localcoord += offset
         seqid, globalcoord = local_to_global(localcoord, target.name)
         snv = VariantSNV(seqid, globalcoord, refr, alt, VW=window,
                          RW=refrwindow, IK=str(len(query.ikmers)))
         snvs.append(snv)
     return snvs
 
 
-def call_deletion(target, query, offset, ksize, leftmatch, indellength):
+def deletion_allele(target, query, offset, ksize, leftmatch, indellength):
     minpos = leftmatch - ksize + 1
     maxpos = leftmatch + ksize - 1
-    window = query.sequence[minpos:maxpos]
+    altwindow = query.sequence[minpos:maxpos]
     minpos += offset
     maxpos += offset + indellength
     refrwindow = target.sequence[minpos:maxpos]
 
     refr = target.sequence[offset+leftmatch-1:offset+leftmatch+indellength]
     alt = refr[0]
-    assert len(refr) == indellength + 1
-    localcoord = offset + leftmatch
-    seqid, globalcoord = local_to_global(localcoord, target.name)
-    return [VariantIndel(seqid, globalcoord - 1, refr, alt, VW=window,
-                         RW=refrwindow, IK=str(len(query.ikmers)))]
+    return refr, alt, refrwindow, altwindow
 
 
-def call_insertion(target, query, offset, ksize, leftmatch, indellength):
+def insertion_allele(target, query, offset, ksize, leftmatch, indellength):
     minpos = leftmatch - ksize + 1
     maxpos = leftmatch + ksize + indellength - 1
-    window = query.sequence[minpos:maxpos]
+    altwindow = query.sequence[minpos:maxpos]
     minpos += offset
     maxpos += offset - indellength
     refrwindow = target.sequence[minpos:maxpos]
 
-    insertion = query.sequence[leftmatch-1:leftmatch+indellength]
-    refr = insertion[0]
-    assert len(insertion) == indellength + 1
-    localcoord = offset + leftmatch
+    alt = query.sequence[leftmatch-1:leftmatch+indellength]
+    refr = alt[0]
+    return refr, alt, refrwindow, altwindow
+
+
+def call_deletion(target, query, offset, ksize, leftmatch, indellength):
```

Comment:
The logic for calling a deletion when the reference locus sequence is shorter than the contig is the same as calling an insertion when the reference is longer. Same idea for calling insertions. So rather than implementing mirror logic in both the `call_deletion` and `call_insertion` sequences, I refactored out the logic to the `insertion_allele` and `deletion_allele` functions. When the first token in the alignment CIGAR string is an `I`, the `call_deletion` function will invoke `insertion_allele` with the contig as reference/target, and the `call_insertion` function will invoke `deletion_allele` with the contig as reference/target.